initrd/sbin/insmod

```diff
@@ -43,9 +43,18 @@ if [ -z "$tpm_missing" ]; then
 	# Extend with the module parameters (even if they are empty) and the
 	# module. Changing the parameters or the module content will result in a
 	# different PCR measurement.
-	tpmr extend -ix "$MODULE_PCR" -ic "$*"
-	tpmr extend -ix "$MODULE_PCR" -if "$MODULE" \
-	|| die "$MODULE: tpm extend failed"
+	if [ -n "$*" ]; then
+		TRACE_FUNC
+		DEBUG "Extending with module parameters and the module's content"
+		tpmr extend -ix "$MODULE_PCR" -ic "$*"
+		tpmr extend -ix "$MODULE_PCR" -if "$MODULE" \
+		|| die "$MODULE: tpm extend failed"
+	else
+		TRACE_FUNC
+		DEBUG "No module parameters, extending only with the module's content"
+		tpmr extend -ix "$MODULE_PCR" -if "$MODULE" \
+		|| die "$MODULE: tpm extend failed"
+	fi
```

Comment:
I don't see any need to have a second code path through here, just more code to test/break.  You can extend a PCR with an empty string, so the above code path works even if the module parameters are empty.

Can we delete this path again?

Reply:
@JonathonHall-Purism then end users/auditors will see a call twice with nothing. If you insist, I can remove, yes.